CoreTesting/src/au/gov/asd/tac/constellation/testing/construction/ImageGraphBuilderPlugin.java

```diff
@@ -134,127 +154,141 @@ protected void edit(final GraphWriteMethods graph, final PluginInteraction inter
+        final int transactionWeightAttributeId = AnalyticConcept.TransactionAttribute.WEIGHT.get(graph);
+
+        final boolean useVertexAttributes = pixelXAttributeId != Graph.NOT_FOUND
+                && pixelYAttributeId != Graph.NOT_FOUND
+                && greenAttributeId != Graph.NOT_FOUND
+                && redAttributeId != Graph.NOT_FOUND
+                && blueAttributeId != Graph.NOT_FOUND
+                && alphaAttributeId != Graph.NOT_FOUND
+                && diffSouthAttributeId != Graph.NOT_FOUND
+                && diffNorthAttributeId != Graph.NOT_FOUND
+                && diffEastAttributeId != Graph.NOT_FOUND
+                && diffWestAttributeId != Graph.NOT_FOUND;
+        final boolean useTransAttributes = transactionWeightAttributeId != Graph.NOT_FOUND;
+
+        int layer = 1;
+        int w = 0;
+        int prevWidth = 0;
+        final boolean addRight = (boolean) parameters.getObjectValue(ADD_RIGHT_PARAMETER_ID);
+        final boolean addLayers = (boolean) parameters.getObjectValue(ADD_LAYERS_PARAMETER_ID);
+        final int stateAttributeId = LayersViewConcept.MetaAttribute.LAYERS_VIEW_STATE.ensure(graph);
+        final LayersViewState currentState = graph.getObjectValue(stateAttributeId, 0);
+        if (currentState != null) {
+            // set start layer to existing graph layer count + 1
+            layer = currentState.getLayerCount() + 1;
+        }
+        
+        for (final BufferedImage image : images) {
 
-            final int vertexVisibilityAttributeId;
-            final boolean multipleFrames = images.size() > 1;
-            if (multipleFrames) {
-                final int attrLayers = graph.addAttribute(GraphElementType.GRAPH, IntegerAttributeDescription.ATTRIBUTE_NAME, "layers", "layers", 0, null);
-                graph.setIntValue(attrLayers, 0, images.size());
-                vertexVisibilityAttributeId = VisualConcept.VertexAttribute.VISIBILITY.get(graph);
-            } else {
-                vertexVisibilityAttributeId = Graph.NOT_FOUND;
+            if (addLayers && layer > 1) {
+                // add new layer for each image if addLayer option selected
+                PluginExecution.withPlugin(new NewLayerPlugin()).executeNow(graph);
+            }
+
+            // If add right option selected, images will be appended to the
+            // right, otherwise, images will be added on top of the prev one
+            if (w > 0 && addRight) {
+                prevWidth = w + prevWidth;
             }
+            w = image.getWidth();
+            final int h = image.getHeight();
+            final int[][] vertexIds = new int[w][h];
+
+            final float zlen = multipleFrames ? 0 : Math.min(w, h) / 4F;
+
+            // get layer bitmask attribute          
+            final int vertexBitmaskAttributeId = LayersConcept.VertexAttribute.LAYER_MASK.ensure(graph);
```

Comment:
similar case here. Probably only want to add this attribute if it is being used (i.e. when `addLayers` is true)